firebase-app-distribution/src/test/java/com/google/firebase/appdistribution/FirebaseAppDistributionTest.java

```diff
@@ -180,4 +201,41 @@ public void signInTester_whenSignInCalledMultipleTimes_cancelsPreviousTask() {
     assertTrue(signInTask1.isCanceled());
     assertFalse(signInTask2.isComplete());
   }
+
+  @Test
+  public void checkForUpdate_whenCalled_getsFidAndAuthToken() throws Exception {
+    Task<AppDistributionRelease> task = firebaseAppDistribution.checkForUpdate();
+    verify(mockFirebaseInstallations, times(1)).getId();
+    verify(mockFirebaseInstallations, times(1)).getToken(true);
+  }
+
+  @Test
+  public void checkForUpdate_whenCalledMultipleTimes_cancelsPreviousTask() throws Exception {
+    Task<AppDistributionRelease> signInTask1 = firebaseAppDistribution.checkForUpdate();
+    Task<AppDistributionRelease> signInTask2 = firebaseAppDistribution.checkForUpdate();
+
+    assertTrue(signInTask1.isCanceled());
+    assertFalse(signInTask2.isComplete());
+  }
+
+  @Test
+  public void getLatestReleaseFromClient_whenNotOnLatestBuild_returnsRelease() {
+    when(mockAppDistributionRelease.getBuildVersion()).thenReturn("3");
```

Comment:
I am probably missing something, but I don't understand these tests. Could you explain to me why we are mocking the app distribution release?

Reply:
This is related to the comment I mentioned above. Since AppDistributionRelease is a simple object, you don't need to use @Mock. You can simply create it using AppDistributionReleaseBuilder

Reply:
I understand now :)